scripts/create-secrets/k8sExecution.js

```diff
@@ -0,0 +1,320 @@
+const childProcess = require("child_process");
+const process = require("process");
+const chalk = require("chalk");
+const trim = require("lodash/trim");
+const {
+    getEnvVarInfo,
+    askIfCreateNamespace,
+    settingNameToEnvVarName
+} = require("./askQuestions");
+const Base64 = require("js-base64").Base64;
+const pwgen = require("./pwgen");
+
+const dbPasswordNames = [
+    "authorization-db",
+    "authorization-db-client",
+    "combined-db",
+    "combined-db-client",
+    "discussions-db",
+    "discussions-db-client",
+    "registry-db",
+    "registry-db-client",
+    "session-db",
+    "session-db-client"
+];
+
+let env = process.env;
+function k8sExecution(config, shouldNotAsk = false) {
+    try {
+        return doK8sExecution(config, shouldNotAsk);
+    } catch (e) {
+        return Promise.reject(e);
+    }
+}
+
+function doK8sExecution(config, shouldNotAsk = false) {
+    env = getEnvByClusterType(config);
+    let configData = Object.assign({}, config.all);
+    const ifAllowEnvVarOverride = configData["allow-env-override-settings"];
+    if (ifAllowEnvVarOverride) {
+        configData = overrideSettingWithEnvVars(configData);
+    }
+    validKubectl();
+    let p = Promise.resolve().then(function() {
+        configData["cluster-namespace"] = trim(configData["cluster-namespace"]);
+        if (!configData["cluster-namespace"]) {
+            throw new Error(
+                "Cluster namespace cannot be empty! \n " +
+                    "If you've set cluster namespace, make sure it's not overrided by env variable."
+            );
+        }
+    });
+    if (!checkNamespace(configData["cluster-namespace"])) {
+        if (shouldNotAsk) {
+            throw new Error(
+                `Namespace ${
+                    configData["cluster-namespace"]
+                } doesn't exist. Please create and try again.`
+            );
+        }
+        p = p
+            .then(
+                askIfCreateNamespace.bind(null, configData["cluster-namespace"])
+            )
+            .then(function(ifCreate) {
```

Comment:
Can we rename `ifCreate` to `shouldCreateNamespace` or something like that?

Reply:
👍